staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -236,6 +238,20 @@ func (t *tracker) List(gvr schema.GroupVersionResource, gvk schema.GroupVersionK
 		return nil, err
 	}
 
+	uList := unstructured.UnstructuredList{}
+	uList.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(list)
+	if err != nil {
+		return nil, err
+	}
+
+	// making sure kind and apiVersion are not lost on creating a new list from scheme
+	uList.SetGroupVersionKind(listGVK)
```

Comment:
rather than working around this in lots of individual places, something like https://github.com/kubernetes/kubernetes/pull/80618/commits/faf14c724f0573ab47568e1a6f2c9a5455cf0305 would be a more systemic fix. I don't think we should complicate this code to work around this issue in one specific place